internal/bminventory/inventory.go

```diff
@@ -3855,14 +3857,6 @@ func (b *bareMetalInventory) CompleteInstallation(ctx context.Context, params in
 		return common.GenerateErrorResponder(err)
 	}
 
-	if b.ocmClient != nil {
-		_, err := b.ocmClient.AccountsMgmt.UpdateSubscription(ctx, cluster.AmsSubscriptionID, cluster.OpenshiftClusterID)
-		if err != nil {
-			log.WithError(err).Errorf("Failed to update AMS subscription for cluster %v", cluster.ID)
-			return common.NewApiError(http.StatusInternalServerError, err)
-		}
-	}
-
```

Comment:
Doesn't it hurts the backward compatibility you mentioned in the PR description?
Updating it twice will work so this is not harmful to keep it.